sdk/go/common/workspace/plugins.go

```diff
@@ -1035,6 +1035,16 @@ func NewPluginSpec(
 			// Prefix the url with `git://`, so we can later recognize this as a git URL.
 			pluginDownloadURL = "git://" + u.String()
 			isGitPlugin = true
+			// If there is no version specified, we version the plugin ourselves. This way the user gets
+			// a consistent experience once the plugin is installed, and won't have any problems when the repo
+			// is updated.
+			if versionStr == "" {
+				var err error
+				version, err = gitutil.GetLatestTagOrHash(context.Background(), url)
```

Comment:
Could you please explain in which flows this will hit? Is it only when calling `pulumi package add` with no version? Or also when running those plugins that were added earlier?

Reply:
It's only when calling `pulumi package add` with no version. Hitting this also means that the `version` will be included in the generated SDK, which in turn means we can find the right version of the plugin again easily/automatically download it again if necessary.  So already downloaded git plugins will know theiri version already, and will never be versionless.

Reply:
Great! Thanks for confirming. Consider adding this as a comment.